rust/ql/lib/codeql/rust/internal/PathResolution.qll

```diff
@@ -359,7 +368,16 @@ private class VariantItemNode extends ItemNode instanceof Variant {
 class FunctionItemNode extends AssocItemNode instanceof Function {
   override string getName() { result = Function.super.getName().getText() }
 
-  override predicate hasImplementation() { super.hasBody() }
+  override predicate hasImplementation() {
+    super.hasBody()
+    or
+    // for trait items from library code, we do not currently know if they
+    // have default implementations or not, so we assume they do
```

Comment:
Why do we assume that they have implementations and not the other way around (assume that they don't)?

Reply:
That is what we did before, and that meant we were unable to resolve such calls

Reply:
Oh, I see. I would've thought that they'd just be prioritized lower in `getStaticTarget`.